cmd/ttn-lw-cli/commands/users_bookmarks.go

```diff
@@ -0,0 +1,203 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.thethings.network/lorawan-stack/v3/cmd/internal/io"
+	"go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/internal/api"
+	"go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/internal/util"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+var (
+	entityIDFlags = util.NormalizedFlagSet()
+
+	errFlagsUnsupported = errors.DefineInvalidArgument("flags_unsupported", "flags are not supported for this command")
+)
+
+func getEntityID(flags *pflag.FlagSet) *ttnpb.EntityIdentifiers {
+	if s, err := flags.GetString("entity-ids.application-id"); s != "" && err == nil {
+		if devID, err := flags.GetString("entity-ids.device-id"); devID != "" && err == nil {
+			return (&ttnpb.EndDeviceIdentifiers{
+				ApplicationIds: &ttnpb.ApplicationIdentifiers{ApplicationId: s},
+				DeviceId:       devID,
+			}).GetEntityIdentifiers()
+		}
+
+		return (&ttnpb.ApplicationIdentifiers{ApplicationId: s}).GetEntityIdentifiers()
+	}
+	if s, err := flags.GetString("entity-ids.client-id"); s != "" && err == nil {
+		return (&ttnpb.ClientIdentifiers{ClientId: s}).GetEntityIdentifiers()
+	}
+	if s, err := flags.GetString("entity-ids.device-id"); s != "" && err == nil {
```

Comment:
Why isn't this equivalent to to the `if` on L36 ? If you provide only the device ID, it should lead to an error, right ?